46.md

```diff
@@ -2,161 +2,83 @@ NIP-46
 ======
 
 Nostr Connect
-------------------------
+-------------
 
 `draft` `optional`
 
-## Rationale
+This NIP describes a method for 2-way communication between a **remote signer** and a normal Nostr client. The remote signer could be, for example, a hardware device dedicated to signing Nostr events, while the client is a normal Nostr client.
 
-Private keys should be exposed to as few systems - apps, operating systems, devices - as possible as each system adds to the attack surface.
+## Signer Discovery
 
-Entering private keys can also be annoying and requires exposing them to even more systems such as the operating system's clipboard that might be monitored by malicious apps.
+The client must somehow be able to contact the signer through a common relay.
 
+### Started by the signer
 
-## Terms
-
-* **App**: Nostr app on any platform that *requires* to act on behalf of a nostr account.
-* **Signer**: Nostr app that holds the private key of a nostr account and *can sign* on its behalf.
+The remote signer generates a connection token in the form
 
+```
+<npub1...>#<secret>?relay=wss://...&relay=wss://...
```

Comment:
Can this also be a URL please? like `nostrconnect://<remote-pubkey>?relay=wss://example.com#<secret>`

this should work in both directions, `metadata` is not required in this direction ofc

Reply:
I don't think there is any both directions. previously the client created the connection details for the signer. now the signer creates the connection details for the client (so its completely reversed). but maybe i'm misunderstanding...

Reply:
Do we even need this `secret`? Maybe we should go on the other direction and just paste an `npub1`?

But yes, I personally prefer the URL, the problem is that doing it will break the the nsecBunker flow as it is implemented in everywhere.

Reply:
In bunker the secret is basically a token which will grant certain access to a specific key, it can expire if not claimed and so on, so it has a purpose there.

Its probably because the client is communicating from a random key and this key is not enrolled into the bunker, they claim the token and then the key is associated to those permissions.

Reply:
I think we should go into a URL, clients are the ones parsing the client, so it's not a breaking change for them.

Reply:
How about `nostrconnect://<nip05|pubkey>?relay=...&secret=...`